src/main/java/org/prebid/server/settings/CompositeApplicationSettings.java

```diff
@@ -39,48 +40,77 @@ private static Proxy createProxy(List<ApplicationSettings> delegates) {
         return proxy;
     }
 
+    /**
+     * Runs a process to get account by id from a chain of retrievers
+     * and returns {@link Future&lt;{@link Account}&gt;}
+     */
     @Override
     public Future<Account> getAccountById(String accountId, Timeout timeout) {
         return proxy.getAccountById(accountId, timeout);
     }
 
+    /**
+     * Runs a process to get AdUnit config by id from a chain of retrievers
+     * and returns {@link Future&lt;{@link String}&gt;}
+     */
     @Override
     public Future<String> getAdUnitConfigById(String adUnitConfigId, Timeout timeout) {
         return proxy.getAdUnitConfigById(adUnitConfigId, timeout);
     }
 
+    /**
+     * Runs a process to get stored requests by a collection of ids from a chain of retrievers
+     * and returns {@link Future&lt;{@link StoredRequestResult}&gt;}
+     */
     @Override
-    public Future<StoredRequestResult> getStoredRequestsById(Set<String> ids, Timeout timeout) {
-        return proxy.getStoredRequestsById(ids, timeout);
+    public Future<StoredRequestResult> getStoredData(Set<String> requestIds, Set<String> impIds, Timeout timeout) {
+        return proxy.getStoredData(requestIds, impIds, timeout);
     }
 
+    /**
+     * Runs a process to get stored requests by a collection of amp ids from a chain of retrievers
+     * and returns {@link Future&lt;{@link StoredRequestResult}&gt;}
+     */
     @Override
-    public Future<StoredRequestResult> getStoredRequestsByAmpId(Set<String> ids, Timeout timeout) {
-        return proxy.getStoredRequestsByAmpId(ids, timeout);
+    public Future<StoredRequestResult> getAmpStoredData(Set<String> requestIds, Set<String> impIds, Timeout timeout) {
+        return proxy.getAmpStoredData(requestIds, Collections.emptySet(), timeout);
     }
 
+    /**
+     * Decorates {@link ApplicationSettings} for a chain of retrievers
+     */
     private static class Proxy implements ApplicationSettings {
 
         private ApplicationSettings applicationSettings;
         private Proxy next;
 
-        Proxy(ApplicationSettings applicationSettings, Proxy next) {
+        /**
+         * Interface to satisfy obtaining {@link StoredRequestResult}
+         *
+         * @param <T> set of stored requests id
+         * @param <U> set of stored imps id
+         * @param <V> processing timeout
+         * @param <R> result of fetching stored data
+         */
+        interface TriFunction<T, U, V, R> {
```

Comment:
Since this interface is used in more than one compilation unit it deserves to become top-level.